### PR TITLE
Bugfix/125

### PR DESCRIPTION
**Pull request by DeepDiver1975**
- implement v-model behavior properly
- apply https://vuejs.org/v2/style-guide/#Private-property-names-essential
- add some more example code
- add public method focus() to set the focus

### CODE REVIEW

**Comment by DeepDiver1975** [comment]
fixes #125

File: src/elements/OcTextInput.vue [hunk 1]
```diff
@@ -5,19 +5,24 @@
...
-    @focus="onFocus($event.target.value)"
+    ref="input"
+    @input="$_ocTextInput_onInput($event.target.value)"
+    @focus="$_ocTextInput_onFocus($event.target.value)"
+    @keydown="$_ocTextInput_onKeyDown($event)"
   />
 </template>
 
 <script>
+import OcButton from "./OcButton"
```
Comment (butonic) [inline src/elements/OcTextInput.vue]: AFAICT the oc-text-input does not need an ocbutton. only your doc bloc does

File: src/elements/OcTextInput.vue [hunk 2]
```diff
@@ -5,19 +5,24 @@
...
+
 /**
  * Form Inputs are used to allow users to provide text input when the expected
  * input is short. Form Input has a range of options and supports several text
  * formats including numbers. For longer input, use the form `Textarea` element.
  */
 export default {
   name: "oc-text-input",
+  components: { OcButton },
```
Comment (butonic) [inline src/elements/OcTextInput.vue]: no, not really

Comment (DeepDiver1975) [reply src/elements/OcTextInput.vue]: I know - could not find an alternative approach
use a plain html button instead?

Comment (butonic) [reply src/elements/OcTextInput.vue]: um I don't think you need to reference it:
```diff
diff --git a/src/elements/OcTextInput.vue b/src/elements/OcTextInput.vue
index 8c85fc6..768e3b6 100644
--- a/src/elements/OcTextInput.vue
+++ b/src/elements/OcTextInput.vue
@@ -110,7 +110,7 @@ export default {
             <h3 class="uk-heading-divider">
                 Interactions
             </h3>
-            <button @click="_focus">Focus input below</button>
+            <oc-button text="Focus input below" @click="_focus" />
             <oc-text-input placeholder="Will you focus on me?" ref="inputForFocus"/>
         </section>
     </template>
```

works just fine for the button

**Pull request merged**